includes/class-amp-story-post-type.php

```diff
@@ -142,6 +142,98 @@ public static function register() {
 		add_filter( 'show_admin_bar', [ __CLASS__, 'show_admin_bar' ] );
 		add_filter( 'replace_editor', [ __CLASS__, 'replace_editor' ], 10, 2 );
 		add_filter( 'admin_body_class', [ __CLASS__, 'admin_body_class' ], 99 );
+		add_filter( 'wp_kses_allowed_html', [ __CLASS__, 'filter_kses_allowed_html' ], 10, 2 );
+
+		// Select the single-amp-story.php template for Stories.
+		add_filter( 'template_include', [ __CLASS__, 'filter_template_include' ] );
+
+		add_action(
+			'amp_story_head',
+			function() {
+				// Theme support for title-tag is implied for stories. See _wp_render_title_tag().
+				echo '<title>' . esc_html( wp_get_document_title() ) . '</title>' . "\n";
+			},
+			1
+		);
+		add_action( 'amp_story_head', 'wp_enqueue_scripts', 1 );
+		add_action(
+			'amp_story_head',
+			function() {
+				/*
+				 * Same as wp_print_styles() but importantly omitting the wp_print_styles action, which themes/plugins
+				 * can use to output arbitrary styling. Styling is constrained in story template via the
+				 * \AMP_Story_Legacy_Post_Type::filter_frontend_print_styles_array() method.
+				 */
+				wp_styles()->do_items();
+			},
+			8
+		);
+
+		add_filter(
+			'amp_content_sanitizers',
+			static function( $sanitizers ) {
+				if ( is_singular( self::POST_TYPE_SLUG ) ) {
+					$sanitizers['AMP_Story_Sanitizer'] = [];
+
+					// Disable noscript fallbacks since not allowed in AMP Stories.
+					$sanitizers['AMP_Img_Sanitizer']['add_noscript_fallback']    = false;
+					$sanitizers['AMP_Audio_Sanitizer']['add_noscript_fallback']  = false;
+					$sanitizers['AMP_Video_Sanitizer']['add_noscript_fallback']  = false;
+					$sanitizers['AMP_Iframe_Sanitizer']['add_noscript_fallback'] = false; // Note that iframe is not yet allowed in an AMP Story.
+				}
+				return $sanitizers;
+			}
+		);
+
+		add_filter(
```

Comment:
Note: this should be handled by the validator, IIRC, needs looking into why the `p` is added/not removed, perhaps there's something missing still.

Reply:
@westonruter For some reason, I remembered as if the Story sanitizer had removing the `p` tag added by `wpautop` as part of it, however, that's not the case. Any reason why it would get removed in the Legacy Story Post (currently in the released plugin) and not in the new Story Post?

Note: added `remove_filter( 'the_content', 'wpautop' );` for now but since it's not necessary for the Legacy Story Post then I suppose there's something missing from the new.

Thoughts?

Reply:
@miina The removal of `wpautop` may be due to the issue where it incorrectly tries to add `p` tags around AMP components. We've had this issue elsewhere when trying to use `<div>` or some other block-level element as a `placeholder` inside of `<amp-iframe>`, which causes it to break. I think it is similar thing here.

In any case, I don't think _any_ of `the_content` filters should apply in the new Story editor. They will just introduce areas for validation errors to occur (see #3788). If any filters are provided, they should be explicitly added one-by-one.

Reply:
> In any case, I don't think any of the_content filters should apply in the new Story editor. They will just introduce areas for validation errors to occur (see #3788). If any filters are provided, they should be explicitly added one-by-one.

Hmm, okay, I guess it's OK then to leave the filter here for the purposes of this PR. Without this filter, the `p` is added and the story displayed blank in the FE due to sanitization.

Generally, the `the_content` filter will probably be addressed/discussed here: #3788

Reply:
What is the benefit of a anonymous function here? Does this make this harder to test?